dockstore-client/src/main/java/io/dockstore/client/cli/nested/AbstractEntryClient.java

```diff
@@ -921,6 +959,115 @@ protected void preValidateLaunchArguments(List<String> args) {
+        ApiClientExtended wesApiClient = new ApiClientExtended();
+        clientWorkflowExecutionServiceApi.setApiClient(wesApiClient);
+
+        INIConfiguration config = Utilities.parseConfig(this.getConfigFile());
+        SubnodeConfiguration configSubNode = null;
+        try {
+            configSubNode = config.getSection("WES");
+        } catch (Exception e) {
+            out("Could not get WES section from config file");
+        }
+
+        String wesEndpointUrl = ObjectUtils.firstNonNull(wesUrl, configSubNode.getString("url"));
+        if (wesEndpointUrl == null || wesEndpointUrl.isEmpty()) {
+            errorMessage("No WES URL found in config file and no WES URL entered on command line. Please add url: <url> to "
+                    + "config file in a WES section or use --wes-url <url> option on the command line", CLIENT_ERROR);
+        } else {
+            out("WES endpoint url is: " + wesEndpointUrl);
+            wesApiClient.setBasePath(wesEndpointUrl);
+        }
+
+        /**
+         * Setup authentication credentials for the WES URL
+         */
+        String wesAuthorizationCredentials = ObjectUtils.firstNonNull(wesCred, configSubNode.getString("authorization"));
+        if (wesAuthorizationCredentials == null) {
+            out("Could not set Authorization header. Authorization key not found in config file and not provided on the command line. "
+                    + "Please add 'authorization: <type> <credentials> to config file in WES section or "
+                    + "use --wes-auth '<type> <credentials>' option on the command line if authorization credentials are needed");
+        } else {
+            wesApiClient.addDefaultHeader(HttpHeaders.AUTHORIZATION, wesAuthorizationCredentials);
+        }
+
+        // TODO Add these headers to the http request. Are these needed?
+        wesApiClient.addDefaultHeader("Accept", "*/*");
+        wesApiClient.addDefaultHeader("Expect", "100-continue");
+
+        clientWorkflowExecutionServiceApi.setApiClient(wesApiClient);
+        return clientWorkflowExecutionServiceApi;
+    }
+
+    /**
+     * Processes Workflow Execution Schema (WES) commands.
+     *
+     * @param args Arguments entered into the CLI
+     */
+    public void processWesCommands(final List<String> args) {
+        if (args.isEmpty() || containsHelpRequest(args)) {
+            launchHelp();
+        } else {
+
+            this.wesUri = optVal(args, "--wes-url", null);
+            this.wesAuth = optVal(args, "--wes-auth", null);
+            if (args.contains("launch")) {
+                out("Launching workflow using WES");
+                //remove the launch keyword so command processing continues as usual
+                args.remove(0);
+                launch(args);
+            } else {
+                WorkflowExecutionServiceApi clientWorkflowExecutionServiceApi = getWorkflowExecutionServiceApi(getWesUri(), getWesAuth());
+                if (args.contains("status")) {
```

Comment:
This could break if the args contain status, but the user mean to call cancel for example. Should check a specific index of the args, based on where the expected subcommand (ex. status) would be.

Reply:
Ex. dockstore wes cancel <id> status

Reply:
OK fixed it to check the first arg in args